src/test/java/org/assertj/core/internal/iterables/Iterables_assertSatisfy_Test.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.internal.iterables;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldSatisfy.shouldSatisfy;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.assertj.core.util.Lists.newArrayList;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.Iterables;
+import org.assertj.core.internal.IterablesBaseTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link Iterables#assertSatisfy(AssertionInfo, Iterable, Consumer[])}</code>.
+ *
+ * @author Ting Sun
+ */
+@DisplayName("Iterables assertSatisfy")
+public class Iterables_assertSatisfy_Test extends IterablesBaseTest {
+
+  private List<String> actual = newArrayList("Luke", "Leia", "Yoda");
+
+  @Test
+  public void should_pass_if_there_is_only_one_consumer_and_can_be_satisfied() {
+    // GIVEN
+    Consumer<String> consumer = s -> {
+      assertThat(s.length()).isEqualTo(4);
+    };
+
+    // WHEN/THEN
+    iterables.assertSatisfy(info, actual, consumer);
+  }
+
+  @Test
+  public void should_pass_if_there_are_multiple_consumers_and_can_be_satisfied() {
+    // GIVEN
+    Consumer<String> consumer1 = s -> {
+      assertThat(s.length()).isEqualTo(4);
+      assertThat(s).doesNotContain("L");
+    };
+    Consumer<String> consumer2 = s -> {
+      assertThat(s).doesNotContain("a");
+    };
+
+    // WHEN/THEN
+    iterables.assertSatisfy(info, actual, consumer1, consumer2);
+  }
+
```

Comment:
Should we add unit tests for element order independence, duplicate elements, and no consumers? 
```suggestion

  @Test
  public void should_pass_there_are_multiple_consumers_and_can_be_satisfied_in_any_order() {
    // GIVEN
    Consumer<String> consumer1 = s -> assertThat(s).contains("L"); // Matches "Luke" and "Leia"
    Consumer<String> consumer2 = s -> assertThat(s).doesNotContain("a"); // Matches "Luke"

    // WHEN/THEN
    iterables.assertSatisfy(info, actual, consumer1, consumer2);
  }
  
  @Test
  public void should_pass_if_iterable_contains_multiple_equal_elements() {
    // GIVEN
    List<String> names = newArrayList("Luke", "Luke");
    Consumer<String> consumer1 = s -> assertThat(s).contains("L");
    Consumer<String> consumer2 = s -> assertThat(s).contains("u");

    // WHEN/THEN
    iterables.assertSatisfy(info, names, consumer1, consumer2);
  }

  @Test
  public void should_pass_if_there_is_no_consumer() {
    // WHEN/THEN
    iterables.assertSatisfy(info, actual);
  }

```